Core/Src/app_threadx.c

```diff
@@ -244,39 +249,41 @@ void thread_1_entry(ULONG thread_input)
 
 
         /* Release the semaphore. */
-        semaphore_status = tx_semaphore_put(&counting_semaphore_0);
+        semaphore_status = tx_semaphore_put(&semaphore_0);
 
         if ( (queue_status != TX_SUCCESS) || (semaphore_status != TX_SUCCESS) )
         {
         	HAL_UART_Transmit(&huart2, (uint8_t*)"Thread 1 Error.\r\n", 17, 100);
         }
-
     }
 }
 
 void thread_2_entry(ULONG thread_input)
 {
 	ULONG received_message;
 	ULONG queue_status;
-
-	HAL_UART_Transmit(&huart2, (uint8_t*)"Made Thread 2.\r\n", 17, 100);
+	ULONG semaphore_status;
 	char message_buffer[100]; // Buffer to store the message string
 
+	HAL_UART_Transmit(&huart2, (uint8_t*)"Made Thread 2.\r\n\r\n", 19, 100);
+
+	/* Signals to thread 1 that thread 2 has been created and the tasks can actually start. */
+	tx_semaphore_put(&semaphore_0);
```

Comment:
I don't believe it is guaranteed that thread 1 will run before thread 2. It might be possible for this to run before thread 1 and deadlock the threads. Something to research.